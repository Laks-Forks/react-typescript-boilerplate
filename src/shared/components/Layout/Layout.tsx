import React from 'react';
import { Helmet } from 'react-helmet';
import useSiteMetaData from 'shared/hooks/useSiteMetaData';

interface IProps {
  children: React.ReactNode;
  title?: string;
  description?: string;
  imageUrl?: string;
}

const Layout = ({ children, title, description, imageUrl }: IProps) => {
  const siteMeta = useSiteMetaData();
  const metaImageUrl = `${siteMeta.url}${imageUrl || siteMeta.imageUrl}`;
  return (
    <div>
      <Helmet>
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1, shrink-to-fit=no"
        />
        <title>{title}</title>
        <meta name="description" content={description} />
        <meta property="og:site_name" content={title} />
        <meta property="og:image" content={metaImageUrl} />
        <meta name="twitter:card" content="summary" />
        <meta name="twitter:title" content={title} />
        <meta name="twitter:description" content={description} />
        <meta name="twitter:image" content={metaImageUrl} />
      </Helmet>
      {children}
    </div>
  );
};

export default Layout;
