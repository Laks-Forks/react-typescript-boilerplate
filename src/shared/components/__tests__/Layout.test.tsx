import React from 'react';
import { Layout } from '../Layout';
import { render } from '@testing-library/react';

describe('Layout', () => {
  it('should match a snapshot', () => {
    const { container } = render(
      <Layout>
        <div>
          <h1>I'm in layout</h1>
        </div>
      </Layout>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
