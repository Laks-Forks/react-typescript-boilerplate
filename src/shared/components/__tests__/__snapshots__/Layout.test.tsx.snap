// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match a snapshot 1`] = `
<div>
  <div>
    <h1>
      I'm in layout
    </h1>
  </div>
</div>
`;
